src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -889,6 +919,18 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         return _terminal->IsTrackingMouseInput();
     }
 
+    // Method Description:
+    // - Handles changes in mouse mode state
+    winrt::fire_and_forget TermControl::_TerminalMouseModeChanged()
+    {
+        co_await Dispatcher();
+        if (_oldCursor) // if we have an active cursor transition
+        {
+            auto coreWindow = Window::Current().CoreWindow();
+            coreWindow.PointerCursor(_CanSendVTMouseInput() ? _pointerCursor : _textCursor);

```

Comment:
Should we have a helper method that's just "set the `PointerCursor` to the correct cursor for our current mouse input mode"?